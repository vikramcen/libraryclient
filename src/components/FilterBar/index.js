import React, {useState, useEffect} from "react";
import { Form } from "react-bootstrap";

export default function FilterBar({filterText, setFilterText}) {

  const handleChange = (e) => {
    const { name, value } = e.target;
    setFilterText({...filterText, [name]: value});
  }

  const handleChange2 = (e) => {
    setFilterText({...filterText, "genre": e.target.value});    
  }

  return(
    <div className="d-flex justify-content-center align-items-center">
      <div className="search-container d-flex justify-content-center flex-column p-2 p-lg-4">
        <form className="d-flex p-md-3 p-lg-0 ">
          <div className="input-container d-flex">
            <input 
              className="form-control mr-sm-2 impForm" 
              name="search"
              type="search" 
              placeholder="Filter keywords" 
              aria-label="Search" 
              onChange={handleChange}
            />
          </div>
        </form>
      </div>
      
      <Form.Group controlId="formBasicSelect">
        <Form.Control
          as="select"
          onChange={handleChange2}
        >
          <option value="">Select Genre</option>
          <option value="fiction">Fiction</option>
          <option value="Biography">Biography</option>
          <option value="Children">Children</option>
        </Form.Control>
      </Form.Group>
    </div>
  )
}