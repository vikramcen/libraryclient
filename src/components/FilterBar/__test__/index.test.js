import React from "react";
import { render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import FilterBar from '..';

afterEach(cleanup);

describe('FilterBar component', () => {
  // First Test
  it('renders', () => {
    render(<FilterBar/>);
  });
  // Second Test
  it('matches snapshot DOM node structure', () => {
    //render About
    const {asFragment} = render(<FilterBar />);
    expect(asFragment()).toMatchSnapshot();
  })
})