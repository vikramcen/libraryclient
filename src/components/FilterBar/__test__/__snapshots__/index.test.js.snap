// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterBar component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center align-items-center"
  >
    <div
      class="search-container d-flex justify-content-center flex-column p-2 p-lg-4"
    >
      <form
        class="d-flex p-md-3 p-lg-0 "
      >
        <div
          class="input-container d-flex"
        >
          <input
            aria-label="Search"
            class="form-control mr-sm-2 impForm"
            name="search"
            placeholder="Filter keywords"
            type="search"
          />
        </div>
      </form>
    </div>
    <div>
      <select
        class="form-control"
        id="formBasicSelect"
      >
        <option
          value=""
        >
          Select Genre
        </option>
        <option
          value="fiction"
        >
          Fiction
        </option>
        <option
          value="Biography"
        >
          Biography
        </option>
        <option
          value="Children"
        >
          Children
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Nav component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center align-items-center"
  >
    <div
      class="search-container d-flex justify-content-center flex-column p-2 p-lg-4"
    >
      <form
        class="d-flex p-md-3 p-lg-0 "
      >
        <div
          class="input-container d-flex"
        >
          <input
            aria-label="Search"
            class="form-control mr-sm-2 impForm"
            name="search"
            placeholder="Filter keywords"
            type="search"
          />
        </div>
      </form>
    </div>
    <div>
      <select
        class="form-control"
        id="formBasicSelect"
      >
        <option
          value=""
        >
          Select Genre
        </option>
        <option
          value="fiction"
        >
          Fiction
        </option>
        <option
          value="Biography"
        >
          Biography
        </option>
        <option
          value="Children"
        >
          Children
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
